org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/context/InjectBundleExtension.java

```diff
@@ -0,0 +1,158 @@
+				setField(field, instance, getBundle(extensionContext, injectBundle));
+			});
+		}
+	}
+
+	/**
+	 * Resolve {@link Parameter} annotated with
+	 * {@link InjectBundleContext @BundleContextParameter} OR
+	 * {@link InjectInstallBundle @InstallBundleParameter} in the supplied
+	 * {@link ParameterContext}.
+	 */
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		Parameter parameter = parameterContext.getParameter();
+		Class<?> parameterType = parameter.getType();
+
+		if (parameterContext.isAnnotated(InjectBundle.class)) {
+			assertParameterIsOfType(parameterType, Bundle.class, InjectBundle.class, ParameterResolutionException::new);
+			InjectBundle injectBundle = parameter.getAnnotation(InjectBundle.class);
+			return getBundle(extensionContext, injectBundle);
+		}
+
+		throw new ExtensionConfigurationException("No parameter types known to BundleContextExtension were found");
+	}
+
+	/**
+	 * Determine if the {@link Parameter} in the supplied
+	 * {@link ParameterContext} is annotated with
+	 * {@link InjectBundleContext @BundleContextParameter} OR
+	 * {@link InjectInstallBundle @InstallBundleParameter}.
+	 */
+	@Override
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		boolean annotatedBundleParameter = parameterContext.isAnnotated(InjectBundle.class);
+
+		if (annotatedBundleParameter && (parameterContext.getDeclaringExecutable() instanceof Constructor)) {
+			throw new ParameterResolutionException(
+				"BundleContextExtension does not support parameter injection on constructors");
+		}
+		return annotatedBundleParameter;
+	}
+
+	private Bundle getBundle(ExtensionContext extensionContext, InjectBundle injectBundle) {
+
+		// return installBundleOf(injectBundle, extensionContext);
+		// TODO:: extra store? Should work with InnstallBundle only.
+		return getStore(extensionContext).getOrComputeIfAbsent(injectBundle,
+			key -> installBundleOf(key, extensionContext), Bundle.class);
+
+	}
+
+	private Bundle installBundleOf(InjectBundle injectBundle, ExtensionContext extensionContext) {
+
+		switch (injectBundle.sourceType()) {
+			case EMBEDDED :
+				return BundleContextExtension.getInstallBundle(extensionContext)
+					.installBundle(injectBundle.value(), injectBundle.start());
+			case URL :
+				try {
+					return BundleContextExtension.getInstallBundle(extensionContext)
+						.installBundle(new URL(injectBundle.value()), injectBundle.start());
+				} catch (MalformedURLException e) {
+					Exceptions.unchecked(() -> e);
+				}
+			case FRAMEWORK :// maybe a earlyer place;
+				return Stream.of(BundleContextExtension.getBundleContext(extensionContext)
+					.getBundles())
+					.filter(b -> b.getSymbolicName()
+						.matches(injectBundle.value()))
+					.findFirst()
+					.orElseThrow(
+						() -> new RuntimeException("No Bundle found with the given Filter: " + injectBundle.value()));
```

Comment:
Wow, that was speedy!

Better to throw `ExtensionConfigurationException` rather than a plain `RuntimeException`.